core/src/main/java/com/datastax/oss/driver/api/core/auth/Authenticator.java

```diff
@@ -68,7 +70,10 @@
    *
    * @param challenge the server's SASL challenge.
    * @return a completion stage that will complete with the updated SASL token (which may be null to
-   *     indicate the client requires no further action).
+   *     indicate the client requires no further action). Note that the driver will <b>clear the
+   *     contents</b> of the returned byte buffer immediately after use (to avoid keeping sensitive
+   *     information in memory); if your authenticator returns the same result for multiple
+   *     evaluations, do not reuse the same buffer.
```

Comment:
You could add: `...do not reuse the same buffer, or make it {@linkplain ByteBuffer#asReadOnlyBuffer() read-only}.`